File: python/exam3/q2/q2.swift
import Foundation

func q2(_ n: Int, _ coords: [[Int]]) -> Int {
    // your code is here
    // ----------------------------------------------
    // ----------------------------------------------
}

func compareOutput(file1: String, file2: String) {
    do {
        let contents1 = try String(contentsOfFile: file1, encoding: .utf8)
        let contents2 = try String(contentsOfFile: file2, encoding: .utf8)
        
        let lines1 = contents1.components(separatedBy: .newlines)
        let lines2 = contents2.components(separatedBy: .newlines)
        
        var success = true
        
        for (i, (line1, line2)) in zip(lines1, lines2).enumerated() {
            if line1 != line2 {
                print("[\(i + 1)] Wrong answer: Yours='\(line1)', Expected='\(line2)'")
                success = false
                break
            }
        }
        
        if success {
            print("Success!")
        }
    } catch {
        print("Error reading files: \(error)")
    }
}

func main() {
    // DO NOT EDIT main() function
    
    var R = 0
    var N = 0
    var result = 0
    
    let start_time = CFAbsoluteTimeGetCurrent()

    FileManager.default.createFile(atPath: "output.txt", contents: nil, attributes: nil)
 
    var str = ""
    
    if let inputURL = Bundle.main.url(forResource: "input", withExtension: "txt"),
       let input = try? String(contentsOf: inputURL, encoding: .utf8),
       let outputURL = Bundle.main.url(forResource: "output", withExtension: "txt") {
        
        let inputLines = input.components(separatedBy: .newlines)
        
        if let r = Int(inputLines[0]) {
            R = r
        }
        if let n = Int(inputLines[1]) {
            N = n
        }

        var coords = Array(repeating: Array(repeating: 0, count: 2), count: N)
        
        for i in 1..<(N + 1) {
            let line = inputLines[i]
            let components = line.components(separatedBy: " ")
            
            if components.count == 2,
                let x = Int(components[0]),
                let y = Int(components[1]) {
                coords[i][0] = x
                coords[i][1] = y
            }
        }
        result = q2(R, coords)
        print(result)
        if str == "" {
            str = String(result)
        }
        else {
            str = str + "\n" + String(result)
        }
        do {
            try str.write(to: outputURL, atomically: true, encoding: .utf8)
        } catch {
            print("File write error")
        }
        
        if let expectedURL = Bundle.main.url(forResource: "expected", withExtension: "txt") {
            compareOutput(file1: outputURL.path, file2: expectedURL.path)
        }
        
        let end_time = CFAbsoluteTimeGetCurrent()
        print("Elapsed time: \(end_time - start_time) seconds.")
    }
    else {
        print("error?")
    }
}

main()

File: python/smu/exam1/exam1/q2/q2.py
import time

def q2(n):
    # your code is here
    # ----------------------------------------------

    memo = {}
    result = []
    for i in range(n):
        if i < 10:
            memo[i] = str(i)
        else:
            memo[i] = chr(55 + i)
    # N = int(input())
    while True:
        if n == 0:
            break
        x = n % 2
        result.append(memo[x])
        n //= 2
    return memo[n]
    # ----------------------------------------------


def compare_output(file1, file2):
    f1 = open(file1, "r")
    f2 = open(file2, "r")

    lines1 = f1.readlines()
    lines2 = f2.readlines()

    success = True
    i = 0
    for _ in lines2:
        if lines1[i].strip() != lines2[i].strip():
            print("[%i] Wrong answer: Yours=\'%s\', Expected=\'%s\'\n" % (i+1, lines1[i].strip(), lines2[i].strip()))
            success = False
            break
        i = i + 1

    if success:
        print("Success!")
    f1.close()
    f2.close()


# main code
input = open("input.txt", "r")      # input data
output = open("output.txt", "w")    # your answer

start_time = time.time()
line = input.readline().split()
K = int(line[0])
for _ in range(K):
    line = input.readline().split()
    n = int(line[0])
    result = q2(n)
    result2 = q2(result)
    if result2 != n:
        print('Error')
    print(result)
    output.write('{}\n'.format(result))

end_time = time.time()
## -----------------------------------------------
    
# DO NOT EDIT. Checking answers
input.close()
output.close()
compare_output("output.txt", "expected.txt")
print('Elapsed time: {:.2f} seconds. '.format(end_time - start_time), end='')
